### ISSUE TITLE
CURLAUTH_DIGEST fails for urls containing query strings when using php+libcurl

### ISSUE THREAD

**Issue by dicksnippe** [curl/curl#3366]
I'm not sure if this is a curl bug or a php bug...
When using libcurl from php (i.e. the curl.so extension in php) digest authentication fails when the url contains a query string. We upgraded curl from 7.61 to 7.62 in php (while keeping php at the same version) and then curl calls from php started failing (for all versions of php we run; php-5.6, 7.1 and 7.2). This is because the uri part in the Authorization header only contains the uri proper, not the query parameters.

Example:
```
$ cat curl2.php
<?php

$url='http://adm.statscollector.omroep.nl/liveprocess.php?param=foo';
$login='user:secret';

$options = array(
        CURLOPT_URL            => $url,
        CURLOPT_USERPWD        => $login,
        CURLOPT_HTTPAUTH       => CURLAUTH_DIGEST,
	CURLINFO_HEADER_OUT	=> true,
);

$ch = curl_init();
curl_setopt_array($ch, $options);

$resp = curl_exec($ch);
$hdrs = curl_getinfo($ch, CURLINFO_HEADER_OUT);
printf("%s\n", $hdrs);
curl_close($ch);

?>
$ php curl2.php
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>400 Bad Request</title>
</head><body>
<h1>Bad Request</h1>
<p>Your browser sent a request that this server could not understand.<br />
</p>
<hr>
<address>Apache/2.4.37 (Unix) OpenSSL/1.1.0i Server at <a href="mailto:webmaster@omroep.nl">adm.statscollector.omroep.nl</a> Port 80</address>
</body></html>
GET /liveprocess.php?param=foo HTTP/1.1
Host: adm.statscollector.omroep.nl
Authorization: Digest username="user", realm="StatsCollectorManager", nonce="B/2S2cV8BQA=29b522210b416ab91a0d1d4cb47d4f0fad4aa9bb", uri="/liveprocess.php", cnonce="OTYzOWMxMzBjOWM1M2NhMmRjMTc3M2RjMmZjODEwMDU=", nc=00000001, qop=auth, response="d46d36edac2b79c6432a16f6f77611a6", algorithm="MD5"
Accept: */*
```
Note how it says uri="/liveprocess.php" instead of "liveprocess.php?param=foo"
Apache reacts with a status 400 and this in the error log:
```
AH01786: uri mismatch - </liveprocess.php> does not match request-uri </liveprocess.php?param=foo>
```

This only happens when php is linked to curl-7.62, not when linked to earlier versions.

However, the thing is that when I compile a small C program to link directly to libcurl and do the same, everything is well, for all versions of curl. Even when linking to the exact same libcurl that't problematic for php.

So it would be reasonable to think that php is using the curl library perhaps in a strange manner(?)
But I can't find any evidence for that assertion.
When hunting for differences between 7.61 and 7.62 this caught the eye: (http.c, line 1869)
```
-  const char *ppath = data->state.path;
+  const char *path = data->state.up.path;
+  const char *query = data->state.up.query;
```
Might that have anything to do with it? Although it doesn't explain why a C program linked against libcurl doesn't havy any problems.

### operating system
Linux

**Comment by dicksnippe** [comment]
Oh dear, I wasn't expecting the markdown formatting. The php sourcecode is a bit of a mess now.
Here it is again, uploaded as .txt
[curl2.php.txt](https://github.com/curl/curl/files/2669584/curl2.php.txt)

**Comment by dicksnippe** [comment]
[stdout.txt](https://github.com/curl/curl/files/2669587/stdout.txt)

**Comment by bagder** [comment]
Duplicate of #3353 Fixed in 552f0205e690 via #3356
